jetty-http2/http2-client/src/test/java/org/eclipse/jetty/http2/client/ConnectTimeoutTest.java

```diff
@@ -50,6 +51,13 @@ public void testConnectTimeout() throws Exception
             @Override
             public void failed(Throwable x)
             {
+                if (x instanceof NoRouteToHostException)
+                {
+                    // Unable to connect to destination is not a failure.
+                    latch.countDown();
+                    return;
+                }
```

Comment:
@joakime I don't know what to say about this. 

We have already called `assumeConnectTimeout()` which verified that connecting to that address _indeed_ throws a `SocketTimeoutException`.

Few lines later we do it via `client.connect()` and we get a different exception?
We should really have `if (docker.screwsUpNetworking())`?

Reply:
Let me try a non-routable, local only, IP instead.

Reply:
The 10.* network is how docker itself communicates among containers.

Reply:
The original test is flaky, even on my local machine (no docker) this test will fail about 15% of the time.
For various reasons: "No Route to Host", "Host unreachable".
I'll use an IANA recommended host instead.